### ISSUE TITLE
CS2 Discussion: Project: Progress as of August 2017: 2.0.0-beta4!

### ISSUE THREAD

**Issue by coffeescriptbot** [jashkenas/coffeescript#4980]
_From @GeoffreyBooth on [2017-08-04 01:33](https://github.com/coffeescript6/discuss/issues/87)_

Hey everyone,

**CoffeeScript 2.0.0-beta4 is released!** Install it via: `npm install coffeescript@next`

Since the [last update](https://github.com/coffeescript6/discuss/issues/86), as of 2017-08-04 the following has been merged into the [jashkenas/coffeescript `2` branch](https://github.com/jashkenas/coffeescript/tree/2) and released in 2.0.0-beta4:

*   [Line comments](https://github.com/jashkenas/coffeescript/pull/4572) (starting with `#`) are now output in the generated JavaScript, thanks to @GeoffreyBooth. You can see them in the [examples in the docs](http://coffeescript.org/v2/#comments).
*   [Block comments](https://github.com/jashkenas/coffeescript/pull/4572) (delimited by `###`) are now allowed anywhere, including inline where they previously weren’t possible, thanks to @GeoffreyBooth. This provides support for **[static type annotations](http://coffeescript.org/v2/#type-annotations)** using Flow’s comments-based syntax. Closes https://github.com/coffeescript6/discuss/issues/12 (at least, via this implementation; feel free to open new issues for other static type solutions you’d like to suggest).
*   Spread syntax (`...` for objects) is [now supported in JSX tags](https://github.com/jashkenas/coffeescript/pull/4607): `<div {props...} />`, thanks to @zdenko.
*   [Argument parsing for scripts run via `coffee` is improved](https://github.com/jashkenas/coffeescript/pull/3946), thanks to @cosmicexplorer. See [breaking changes](http://coffeescript.org/v2/#breaking-changes-argument-parsing-and-shebang-lines). This PR was started in 2015!
*   CLI: [Propagate `SIGINT` and `SIGTERM` signals](https://github.com/jashkenas/coffeescript/pull/4625) when node is forked, thanks to @Gobie.
*   [`await` in the REPL is now allowed](https://github.com/jashkenas/coffeescript/pull/4604) without requiring a wrapper function, thanks to @connec.
*   [`do super` is now allowed](https://github.com/jashkenas/coffeescript/pull/4627), thanks to @zdenko.
*   [Multiple accesses of `super`](https://github.com/jashkenas/coffeescript/pull/4592) like `super.x.y` or `super['x'].y` now work, thanks to @helixbass.
*   Splat/spread syntax [triple dots are now allowed on either the left or the right](https://github.com/jashkenas/coffeescript/pull/4606) (so `props...` or `...props` are both valid), thanks to @zdenko.
*   [Tagged template literals are recognized as callable functions](https://github.com/jashkenas/coffeescript/pull/4601), thanks to @helixbass.
*   Bugfixes for [object spread syntax in nested properties](https://github.com/jashkenas/coffeescript/pull/4611), thanks to @zdenko.
*   Bugfixes for [destructured function parameter default values](https://github.com/jashkenas/coffeescript/pull/4596), thanks to @zdenko.
*   [Fix regressions](https://github.com/jashkenas/coffeescript/pull/4599) related to [chained function calls](https://github.com/jashkenas/coffeescript/pull/4605) and indented `return` and `throw` arguments, thanks to @helixbass.
*   The [REPL no longer warns](https://github.com/jashkenas/coffeescript/pull/4559) about assigning to `_`, thanks to @xixixao.

There are no breaking-change pull requests in the pipeline. This means that unless people report any showstopper bugs with this release, **this could be the last beta before 2.0.0 is released.** So please try it in your projects and let us know if it’s ready. The sooner we get feedback saying that all is well (or that there are bugs we need to address) the sooner we can release 2.0.0.

If you encounter any issues, after first reviewing the [breaking changes](http://coffeescript.org/v2/#breaking-changes), please [open an issue](https://github.com/jashkenas/coffeescript/issues) with “[CS2]“ in the title. Please _don’t_ comment bug reports on this thread. Thanks!

**Comment by coffeescriptbot** [comment]
_From @GeoffreyBooth on [2017-08-14 20:03](https://github.com/coffeescript6/discuss/issues/87#issuecomment-322293975)_


So, any reports of projects working well under 2.0.0-beta4?

As far as I can tell, we have two open bugs that need fixing before we can release 2.0.0, and there are WIP pull requests for both of them. See https://github.com/jashkenas/coffeescript/milestone/10. Now would be the time to let us know if there are any other issues that should be addressed before the 2.0.0 release.

The other task that I’d like to get addressed soonish, ideally before the 2.0.0 release, is https://github.com/jashkenas/coffeescript/issues/4615: demo projects showing how to use CoffeeScript 2 with Webpack, Gulp, Babel etc. Even if you know nothing about the CoffeeScript codebase, if you can spare the time to whip up an example repo it would make a great addition to our documentation!

**Comment by coffeescriptbot** [comment]
_From @objectkit on [2017-08-15 15:16](https://github.com/coffeescript6/discuss/issues/87#issuecomment-322498130)_


Looking forward to testing out later and will post at links - new.target needs to be in there IMO

**Comment by coffeescriptbot** [comment]
_From @GeoffreyBooth on [2017-08-24 01:56](https://github.com/coffeescript6/discuss/issues/87#issuecomment-324510551)_


@objectkit We would welcome a PR 😄
